From 1b29fb957438d76a22ef10fbc58d21c12b8c4d83 Mon Sep 17 00:00:00 2001
From: "Jason M. Bills" <jason.m.bills@intel.com>
Date: Tue, 5 Apr 2022 16:46:12 -0700
Subject: [PATCH] Suppress deprecation warnings for OpenSSL 3.0

When updating to OpenSSL 3.0, deprecated APIs begin throwing warnings.
The APIs are still supported for now, so to give us time to update, just
suppress the warnings for now to allow updating to OpenSSL 3.0.

Tested:
Confirmed that optee-test builds with OpenSSL 3.0

Signed-off-by: Jason M. Bills <jason.m.bills@intel.com>
---
 host/xtest/regression_8100.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/host/xtest/regression_8100.c b/host/xtest/regression_8100.c
index 78b26fb..2388bee 100644
--- a/host/xtest/regression_8100.c
+++ b/host/xtest/regression_8100.c
@@ -128,7 +128,12 @@ static void osslerr(void)
 		const char *f = NULL;
 		int l = 0;
 
+    // TODO: ERR_get_error_line() is deprecated in OpenSSL 3.0 and needs to be
+    // updated
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 		e = ERR_get_error_line(&f, &l);
+#pragma GCC diagnostic pop
 		if (!e)
 			return;
 		ERR_error_string_n(e, b, sizeof(b));
-- 
2.25.1

